base/util.jl

```diff
@@ -220,6 +225,12 @@ function julia_cmd(julia=joinpath(Sys.BINDIR, julia_exename()))
     if opts.use_sysimage_native_code == 0
         push!(addflags, "--sysimage-native-code=no")
     end
+    if opts.use_pkgimage_native_code == 0
+        push!(addflags, "--pkgimage-native-code=no")
+    else
+        # If pkgimage is set, malloc_log and code_coverage should not
+        @assert opts.malloc_log == 0 && opts.code_coverage == 0
```

Comment:
This is an actual assertion. This checks that no-one tried to mess with the options after startup.

If you try to set it on the command line it will throw an appropriate error:
```
./julia --pkgimage-native-code=yes --track-allocation
ERROR: julia: Can't use --pkgimage-native-code=yes together with --track-allocation or --code-coverage.
```